waiter/src/waiter/work_stealing.clj

```diff
@@ -123,33 +128,35 @@
                          ([_]
                            (let [router-id->metrics (service-id->router-id->metrics service-id)
                                  _ (log/trace label "received metrics from" (count router-id->metrics) "routers")
-                                 slots-offered (count request-id->work-stealer)
                                  extra-slots (-> (router-id->metrics router-id)
-                                                 (assoc "slots-offered" slots-offered)
                                                  (utils/compute-help-required)
                                                  (unchecked-negate))
+                                 offered-slots (count request-id->work-stealer)
+                                 stealable-slots (- extra-slots offered-slots)
```

Comment:
Our clusters are operating with degraded performance if router A is using borrowed slots from router B, and vice versa.  Do we have reason to believe this occurs, and what reasonable mechanism, if any, can we put in place to prevent it?

Reply:
This can occur:
1. A receives requests more than its available slots
2. B offers help to A and A starts using these instances
3. Requests now show up in B, but B has no more available instances
4. Requests at A which were using slots assigned to A complete (while the offered instances are still chugging along).
5. A offers its slots as help to B
6. B starts using these instances.

I'm not sure this is entirely bad as otherwise requests that showed up in A would have to wait longer to get an instance.